include/ustore/cpp/types.hpp

```diff
@@ -78,11 +78,25 @@ inline collection_key_field_t ckf(args_at&&... args) {
 /**
  * @brief Graph Edge, or in DBMS terms - a relationship.
  */
+inline ustore_key_t edge_id(size_t a, size_t b) {
+    return (a + b) * (a + b + 1) / 2 + a;
+}
+
 struct edge_t {
     ustore_key_t source_id;
     ustore_key_t target_id;
     ustore_key_t id {ustore_default_edge_id_k};
 
+    inline edge_t() = default;
+    inline edge_t(edge_t&&) = default;
+    inline edge_t(edge_t const&) = default;
+    inline edge_t& operator=(edge_t&&) = default;
+    inline edge_t& operator=(edge_t const&) = default;
+    inline edge_t(ustore_key_t source_id, ustore_key_t target_id) noexcept
+        : source_id(source_id), target_id(target_id) {
+        id = edge_id(source_id, target_id);
```

Comment:
This entirely changes the behavior of Graph collections. Why should we adopt that?